packages/core/src/components/tooltip/tooltip.tsx

```diff
@@ -134,6 +149,14 @@ export class Tooltip<
                     },
                 }}
                 {...restProps}
+                // eslint-disable-next-line react/jsx-no-bind
+                renderTarget={renderTarget ? props => renderTarget({ ...props, tooltipId }) : undefined}
+                content={
+                    // want Popover to warn if empty, so don't wrap in element if empty.
+                    Utils.isReactNodeEmpty(content)
+                        ? content
+                        : Utils.ensureElement(content, undefined, { role: "tooltip", id: tooltipId })
```

Comment:
I don't think we can do this because a consumer may have already defined an `id` on the `content` which this would override

Reply:
Highly doubt that would ever be the case, can't imagine what other scenarios they'd be doing this for. There are many cases, such as in `Popover`, where we apply a bunch of props without caring what the consumer may have already applied.

Reply:
This should account for that case: https://github.com/palantir/blueprint/pull/6865/commits/be12eb309882e8251c8d6e81a170c20f9904cfb0

Reply:
@evansjohnson  the other alternative solution here would be we add a prop `contentProps` to `Popover` and apply it to the `div` that's already wrapping the `content`. Then we could apply these props via that. Up to you.

Reply:
Refactored, please re review, I think this concern has been solved